### ISSUE TITLE
Setting custom Content classes in non legacy stack

### ISSUE THREAD

**Issue by ToBe998** [bolt/bolt#7287]
It is currently possible to create your own `Content` class for specific content types and configure Bolt to use them via the `class` option in your contenttypes.yml.

Common use cases are:
- Override the `link()` method to create special urls for your content records.
- Override the `getImage()` or `getExcerpt()` methods to support custom fields from other extensions
- Override `getRelation()` to imlement faster methods for complex relations in larger website via caching or custom selects.

This works great for anything that uses the legacy storage system, wich is still used in twig templates and some older parts of Bolt. Since more and more parts are slowly switched over to the new storage stack, we need a similar solution for this, that is usable in both stacks simultaneously (since Bolt uses both simultaneously).

**Requirements**
- Works in legacy Content class
- Works in new Content class
- Allows to change Content class to a different one (with same interface) in all parts of Bolt (Back/Front/Twig)

**Comment by rossriley** [comment]
It is doable now but you need to provide your own Repository class to do it...

The default repo is defined by default as:

```
$app['storage.repository.default'] = 'Bolt\Storage\Repository\ContentRepository';
```

If you replace that with a pointer to your custom class (and just make the class extend the ContentRepository class)

Then the only method you need to change in your new class is this one that's in there.

```
     /**
     * @return string
     */
    public function getEntityName()
    {
        return $this->entityName;
    }
```

By default that will return the `Bolt\Storage\Entity\Content` but you can change that to return the fully qualified name to your replacement.

**Comment by ToBe998** [comment]
Is something planned to make this easier after legacy has been removed entirely?

**Comment by rossriley** [comment]
yes, I don't see any reason why we can't continue to support the `class` config on contentypes for when we switch over.

**Comment by bobdenotter** [comment]
This has been fixed by Ross for 3.6, in #7571
